src/Compilers/CSharp/Portable/Lowering/ExtensionMethodReferenceRewriter.cs

```diff
@@ -117,7 +117,7 @@ static BoundExpression updateCall(
 
                     invokedAsExtensionMethod = true;
 
-                    Debug.Assert(receiverOpt.Type!.Equals(method.Parameters[0].Type, TypeCompareKind.ConsiderEverything));
+                    Debug.Assert(receiverOpt.Type!.Equals(method.Parameters[0].Type, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes));
```

Comment:
>IgnoreNullableModifiersForReferenceTypes [](http://example.com/codeflow?start=101&length=40)

```IgnoreAllOptions```?

Reply:
That's not necessary. We create a conversion in `CreateConversion` when there are non-nullability differences.